docs/resources/deployment.md

```diff
@@ -80,6 +81,16 @@ resource "vercel_deployment" "example" {
 - **id** (String) The ID of this resource.
 - **url** (String) A unique URL that is automatically generated for a deployment.
 
+<a id="nestedatt--git_source"></a>
+### Nested Schema for `git_source`
+
+Optional:
+
+- **ref** (String) Branch or commit hash
+- **repo_id** (String) Frontend git repo ID
+- **type** (String) Type of git repo, supported values are: github
```

Comment:
How are these all optional?

Reply:
I set them as required, let me check why they appear as optional in the docs

Reply:
I was investigating a bit and could not find why this happens. tfplugindocs just renders them as optional even when they are set as required. if happens the same [here](https://github.com/vercel/terraform-provider-vercel/blob/main/docs/resources/deployment.md#read-only)

Reply:
Ahh interesting. I guess it's because the parent property is optional then.